From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: LoJoSho <contact@lojosho.com>
Date: Sun, 17 Jul 2022 22:47:25 -0500
Subject: [PATCH] Remove food calculation


diff --git a/src/main/java/net/minecraft/world/food/FoodData.java b/src/main/java/net/minecraft/world/food/FoodData.java
index 2934b6de1f1fb914a532ee20184df99d1acd8e65..39c89005aa85246f604a049c9a0662340aee2a52 100644
--- a/src/main/java/net/minecraft/world/food/FoodData.java
+++ b/src/main/java/net/minecraft/world/food/FoodData.java
@@ -57,8 +57,10 @@ public class FoodData {
     }
 
     public void tick(Player player) {
-        Difficulty enumdifficulty = player.level.getDifficulty();
-
+        // Cherrybomb Start
+        this.tickTimer = 0;
+        this.setFoodLevel(20);
+        /*
         this.lastFoodLevel = this.foodLevel;
         if (this.exhaustionLevel > 4.0F) {
             this.exhaustionLevel -= 4.0F;
@@ -109,7 +111,8 @@ public class FoodData {
         } else {
             this.tickTimer = 0;
         }
-
+        */
+        // Cherrybomb end
     }
 
     public void readAdditionalSaveData(CompoundTag nbt) {
